## **Lab 3 - Naive Bayes**
### S Kailash 18MIS1074 
### SWE4012 Machine Learning (L11+L12)
### Faculty : Dr. M Premlatha
### Date : 1st February 2022

In [27]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import CategoricalNB
from sklearn.datasets import load_iris
from sklearn.datasets import load_wine
from sklearn import datasets
from sklearn.naive_bayes import GaussianNB
from sklearn import naive_bayes
from sklearn.metrics import accuracy_score
from sklearn.preprocessing import LabelEncoder, StandardScaler
from sklearn import metrics

Describing the dataframes

In [28]:
iris=datasets.load_iris()
df=pd.read_csv('Social_Network_Ads.csv')
wine=datasets.load_wine()

print("Shape:",iris.data.shape)
print("Features:",iris.feature_names)
print("Output:",iris.target_names)

print("Shape:",sna.shape)

print("Shape:",wine.data.shape)
print("Features:",wine.feature_names)
print("Output:",wine.target_names)

Shape: (150, 4)
Features: ['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)']
Output: ['setosa' 'versicolor' 'virginica']
Shape: (400, 5)
Shape: (178, 13)
Features: ['alcohol', 'malic_acid', 'ash', 'alcalinity_of_ash', 'magnesium', 'total_phenols', 'flavanoids', 'nonflavanoid_phenols', 'proanthocyanins', 'color_intensity', 'hue', 'od280/od315_of_diluted_wines', 'proline']
Output: ['class_0' 'class_1' 'class_2']


In [29]:
X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)
print(X_train.shape)
print(y_test.shape)

(120, 4)
(30,)


In [30]:
cnb1=CategoricalNB()
cnb1.fit(X_train,y_train)
y_predict = cnb1.predict(X_test)
print(y_predict)

[2 1 0 2 0 2 0 1 1 1 2 1 1 1 1 0 1 1 0 0 1 1 0 0 1 0 0 1 1 0]


In [31]:
print('Accucary:', metrics.accuracy_score(y_test, y_predict))
print('Confusion Matrix\n',metrics.confusion_matrix(y_test, y_predict))


Accucary: 0.9333333333333333
Confusion Matrix
 [[11  0  0]
 [ 0 13  0]
 [ 0  2  4]]


### Here, we will be doing preprocessing and label encoding for the social network advertising dataset.

In [32]:
X=df.iloc[:,[1,2,3]].values
y=df.iloc[:,-1].values
le=LabelEncoder()
X[:,0] = le.fit_transform(X[:,0])
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.20, random_state = 42)
sc = StandardScaler()
X_train = sc.fit_transform(X_train)
X_test = sc.transform(X_test)

In [33]:
classifier = GaussianNB()
classifier.fit(X_train, y_train)

GaussianNB()

In [34]:
y_pred=classifier.predict(X_test)
y_pred
ac = accuracy_score(y_test,y_pred)
print('Accuracy',ac)

Accuracy 0.9375


In [35]:
wine=datasets.load_wine()
X=wine.data
y=wine.target
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25)
print(X_train.shape)
print(y_test.shape)

(133, 13)
(45,)


In [36]:
bnb=naive_bayes.MultinomialNB()
bnb.fit(X_train,y_train)
y_pred=bnb.predict(X_test)
ac = accuracy_score(y_test,y_pred)
print('Accuracy',ac)

Accuracy 0.8


## Therefore, the results are as follows for the 3 datasets :

### Iris dataset - 0.933
### Social Advertisements dataset - 0.937
### Wine dataset - 0.8